src/mod/applications/mod_avmd/mod_avmd.c

```diff
@@ -827,7 +827,7 @@ static void avmd_fire_event(enum avmd_event type, switch_core_session_t *fs_s, d
         return;
     }
 
-    switch_core_session_queue_event(fs_s, &event);
+    switch_core_session_queue_event(fs_s, &event_copy);
```

Comment:
Both switch_core_session_queue_event and switch_fire_event will publish the event. If the switch_core_session_queue_event is successful then we don't need to call switch_fire_event, otherwise we immediately fire the event using switch_fire_event

Reply:
Good point. 
The above change will actually work though since `switch_core_session_queue_event` sets event data to NULL if successful thus, `switch_event_fire` won't publish any event. But I guess its safe to just have a function publish the event.

Reply:
`switch_core_session_queue_event` and `switch_event_fire` are using different queues. Both are needed for certain things.
 - `switch_core_session_queue_event` uses session queue (the one used for DTMF for example, IVR events)
 - `switch_event_fire` operates on global event dispatch queue (e.g. firing text messages through event socket)
 It is intentional to have both calls in there, because removing e.g. `switch_core_session_queue_event` for some events, may affect modules.

But, If this makes FS to send event twice then `switch_core_session_queue_event` should be checked.

On the other hand, there's likely no modules at Telnyx depending on reading AVMD events from session.
So, if you want a quick fix, I suggest removing `switch_core_session_queue_event`.